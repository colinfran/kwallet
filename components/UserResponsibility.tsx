import React, { useContext } from "react"
import { StyleSheet, View, Dimensions } from "react-native"
import { Box, Text, Checkbox } from "native-base"
import { DataContext } from "../providers/DataProvider"

const UserResponsibility = ({
  checkbox1,
  checkbox2,
  checkbox3,
  onCheckboxPress,
}): JSX.Element => {
  const { appColor } = useContext(DataContext)
  return (
    <View style={{ marginTop: 30 }}>
      <View>
        <Box bg="gray.100" rounded="xl" style={styles.boxCheckbox}>
          <View style={styles.boxCheckboxView}>
            <Box style={{ gap: 40 }} width="100%">
              <Checkbox
                _checked={{
                  backgroundColor: appColor,
                  borderColor: appColor,
                }}
                _text={{ width: "100%" }}
                backgroundColor="gray.100"
                isChecked={checkbox1}
                testID="checkbox1"
                value="one"
                width={"100%"}
                onChange={() => onCheckboxPress(1)}
              >
                <Text style={styles.checkboxText}>
                  If I lose my secret phrase, I will not be able to recover this
                  wallet
                </Text>
              </Checkbox>
              <Checkbox
                _checked={{
                  backgroundColor: appColor,
                  borderColor: appColor,
                }}
                _text={{ width: "100%" }}
                backgroundColor="gray.100"
                isChecked={checkbox2}
                testID="checkbox2"
                value="two"
                width={"100%"}
                onChange={() => onCheckboxPress(2)}
              >
                <Text style={styles.checkboxText}>
                  I am responsible for any issue that happens when using this
                  wallet.
                </Text>
              </Checkbox>
              <Checkbox
                _checked={{
                  backgroundColor: appColor,
                  borderColor: appColor,
                }}
                _text={{ width: "100%" }}
                backgroundColor="gray.100"
                isChecked={checkbox3}
                testID="checkbox3"
                value="three"
                width={"100%"}
                onChange={() => onCheckboxPress(3)}
              >
                <Text style={styles.checkboxText}>
                  I am ready to create my new wallet.
                </Text>
              </Checkbox>
            </Box>
          </View>
        </Box>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  boxLogo: {
    flex: 1,
    width: "100%",
    height: Dimensions.get("window").height / 3,
    justifyContent: "center",
    alignItems: "center",
  },
  boxCheckbox: {
    width: "100%",
  },
  boxCheckboxView: {
    justifyContent: "flex-start",
    alignItems: "flex-start",
    gap: 30,
    padding: 20,
  },
  checkboxText: {
    flexWrap: "wrap",
    paddingLeft: 20,
    paddingRight: 20,
    flexGrow: 1,
    flexShrink: 1,
    color: "#000",
  },
})

export default UserResponsibility
