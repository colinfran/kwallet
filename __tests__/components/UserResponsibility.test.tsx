import React from "react"
import { Providers } from "../helper"
import { fireEvent, render, screen } from "@testing-library/react-native"
import UserResponsibility from "../../components/UserResponsibility"

const mockFunction = jest.fn()

describe("<UserResponsibility />", () => {
  it("renders correctly", () => {
    const tree = render(
      <UserResponsibility
        checkbox1={false}
        checkbox2={false}
        checkbox3={false}
        onCheckboxPress={mockFunction}
      />,
      { wrapper: Providers }
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("text", () => {
    render(
      <UserResponsibility
        checkbox1={false}
        checkbox2={false}
        checkbox3={false}
        onCheckboxPress={mockFunction}
      />,
      { wrapper: Providers }
    )
    const kwallet = screen.getByText("I am ready to create my new wallet.")
    expect(kwallet).toBeDefined()
  })

  it("checkboxes", async () => {
    render(
      <UserResponsibility
        checkbox1={false}
        checkbox2={false}
        checkbox3={false}
        onCheckboxPress={mockFunction}
      />,
      { wrapper: Providers }
    )
    const checkbox1 = screen.getByTestId("checkbox1")
    fireEvent.press(checkbox1)
    const checkbox2 = screen.getByTestId("checkbox2")
    fireEvent.press(checkbox2)
    const checkbox3 = screen.getByTestId("checkbox3")
    fireEvent.press(checkbox3)
    expect(mockFunction.mock.calls.length).toBe(3)
  })
})
