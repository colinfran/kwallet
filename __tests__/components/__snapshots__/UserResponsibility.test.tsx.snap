// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserResponsibility /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "marginTop": 30,
      }
    }
  >
    <View>
      <View
        dataSet={{}}
        style={
          [
            {
              "backgroundColor": "#f4f4f5",
              "borderRadius": 12,
            },
            {
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "flex-start",
              "gap": 30,
              "justifyContent": "flex-start",
              "padding": 20,
            }
          }
        >
          <View
            dataSet={{}}
            style={
              [
                {
                  "width": "100%",
                },
                {
                  "gap": 40,
                },
              ]
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": false,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={{}}
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <View
                    checked={false}
                    colorScheme="primary"
                    dataSet={{}}
                    defaultIsChecked={false}
                    disabled={false}
                    isChecked={false}
                    onChange={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#f4f4f5",
                        "borderColor": "#737373",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "display": "flex",
                        "justifyContent": "center",
                        "opacity": 1,
                        "paddingBottom": 2,
                        "paddingLeft": 2,
                        "paddingRight": 2,
                        "paddingTop": 2,
                      }
                    }
                    testID="checkbox1"
                    value="one"
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "color": "#171717",
                          "height": 12,
                          "width": 12,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 8,
                    }
                  }
                />
                <Text
                  dataSet={{}}
                  style={
                    [
                      {
                        "backgroundColor": undefined,
                        "color": "#fafafa",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      },
                      {
                        "color": "#000",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "flexWrap": "wrap",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                      },
                    ]
                  }
                >
                  If I lose my secret phrase, I will not be able to recover this wallet
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": false,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={{}}
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <View
                    checked={false}
                    colorScheme="primary"
                    dataSet={{}}
                    defaultIsChecked={false}
                    disabled={false}
                    isChecked={false}
                    onChange={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#f4f4f5",
                        "borderColor": "#737373",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "display": "flex",
                        "justifyContent": "center",
                        "opacity": 1,
                        "paddingBottom": 2,
                        "paddingLeft": 2,
                        "paddingRight": 2,
                        "paddingTop": 2,
                      }
                    }
                    testID="checkbox2"
                    value="two"
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "color": "#171717",
                          "height": 12,
                          "width": 12,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 8,
                    }
                  }
                />
                <Text
                  dataSet={{}}
                  style={
                    [
                      {
                        "backgroundColor": undefined,
                        "color": "#fafafa",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      },
                      {
                        "color": "#000",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "flexWrap": "wrap",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                      },
                    ]
                  }
                >
                  I am responsible for any issue that happens when using this wallet.
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": false,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={{}}
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  />
                  <View
                    checked={false}
                    colorScheme="primary"
                    dataSet={{}}
                    defaultIsChecked={false}
                    disabled={false}
                    isChecked={false}
                    onChange={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#f4f4f5",
                        "borderColor": "#737373",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "display": "flex",
                        "justifyContent": "center",
                        "opacity": 1,
                        "paddingBottom": 2,
                        "paddingLeft": 2,
                        "paddingRight": 2,
                        "paddingTop": 2,
                      }
                    }
                    testID="checkbox3"
                    value="three"
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "color": "#171717",
                          "height": 12,
                          "width": 12,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 8,
                    }
                  }
                />
                <Text
                  dataSet={{}}
                  style={
                    [
                      {
                        "backgroundColor": undefined,
                        "color": "#fafafa",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      },
                      {
                        "color": "#000",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "flexWrap": "wrap",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                      },
                    ]
                  }
                >
                  I am ready to create my new wallet.
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
